dap_prinz_green_jobs/pipeline/ojo_application/extract_green_measures_flow.py

```diff
@@ -0,0 +1,160 @@
+"""
+Extract green measures pipeline
+--------------
+A pipeline that extracts green measures on a OJO sample
+    as defined in dap_prinz_green_jobs/getters/data_getters/ojo.py
+
+It also saves the extracted green measures to s3.
+
+python dap_prinz_green_jobs/pipeline/ojo_application/extract_green_measures_flow.py run
+"""
+from dap_prinz_green_jobs.pipeline.green_measures.green_measures import GreenMeasures
+from dap_prinz_green_jobs.getters.ojo_getters import (
+    get_ojo_skills_sample,
+    get_ojo_sample,
+    get_ojo_job_title_sample,
+)
+
+from dap_prinz_green_jobs import logger
+
+from toolz import partition_all
+from metaflow import FlowSpec, step, Parameter
+
+# instantiate GreenMeasures class here
+gm = GreenMeasures()
+
+
+class ExtractGreenMeasuresFlow(FlowSpec):
+    production = Parameter("production", help="Run in production?", default=False)
+    batch_size = Parameter("batch_size", help="Batch size", default=5000)
+
+    @step
+    def start(self):
+        """
+        Start step
+        """
+        self.next(self.load_ojo_sample)
+
+    @step
+    def load_ojo_sample(self):
+        """
+        loads ojo skills sample from s3 and reformat it to be compliant with GreenMeasures
+        """
+        import dap_prinz_green_jobs.pipeline.green_measures.skills.skill_measures_utils as sm
+
+        # also load extracted skills so as not to duplicate skills extraction
+        logger.info("loading and reformattting extracted skills from ojo sample...")
+        ojo_skills_raw = get_ojo_skills_sample()
+
+        if not self.production:
+            ojo_skills_raw = ojo_skills_raw[:1]
+
+        # reformat it to be the output of es.get_skills() for GreenMeasures
+        self.ojo_skills = (
+            ojo_skills_raw.groupby("id")
+            .skill_label.apply(list)
+            .reset_index()
+            .assign(skills_formatted=lambda x: x.skill_label.apply(sm.format_skills))
+            .skills_formatted.tolist()
+        )
+
+        # chunk job skills
+        logger.info("chunking already extracted job skills...")
+
+        if self.production:
+            self.job_skill_chunks = list(
+                partition_all(self.batch_size, self.ojo_skills)
+            )
+        else:
+            self.job_skill_chunks = list(partition_all(1, self.ojo_skills))
+
+        self.next(self.extract_skill_measures, foreach="job_skill_chunks")
```

Comment:
a way to speed things up might be to get the unique skill entities and just map those (I imagine you are mapping 1000s of "communication skills" separately - unless I've forgotten how the skills extractor works?), but you'd then have to be careful to map them back to the relevant job adverts